### ISSUE TITLE
Proxies cannot proxy local repository

### ISSUE THREAD

**Issue by MiniDay** [dzikoysk/reposilite#472]
I am a minecraft developer. I use Maven to develop minecraft bukkit plugin. Sometimes I upload some of my own code to my Maven repository for others to use.

I used to use nexus as my Maven repository manager, but when I knew about repository, I began to use it all the time.

Repository is really lightweight and suitable for individual developers like me.

However, minecraft developers come from all over the world. Because the location of Maven warehouse is too far away from them, it often happens that the dependency download time is too long (even unable to download).

So generally, I set up three repositories for my Maven repository manager

repository A：upload myself artifact

repository B: proxy to bukkit (and others) repository

repository C: contains A and B

In this way, when I use it myself, I just need to add the URL of repository C, so that I can ensure the speed (and stability) of downloading bukkit dependency, and I can also use my other artifacts at the same time.

When other people use my repository, they only need to add repository a, which can ensure that they will only download my code from my repository, and will not include the bukkit (or others) code. In this way, even if you are in different countries and geographical locations, you can try not to be affected by the network quality of my warehouse（ At the same time, it can also save traffic for my server.)

I'm sorry it's a lot of nonsense, but that's what I'm really going to say.

When I use repository, I find that I can't set it to the ideal state.

i set two repositories:
```
repositories [
  maven-public
  hamster-repo
]
```

and i added proxies:
```
proxied [
  https://hub.spigotmc.org/nexus/content/repositories/snapshots
]
```

then i deployed my artifact into hamster-repo

and i write pom.xml with:
```xml
    <repositories>
        <repository>
            <id>hamster-public</id>
            <url>http://maven.hamster3.cn/maven-public</url>
        </repository>
    </repositories>
```
finally i can download bukkit dependency, but i can't download myself artifact.

unless i use maven settings with:
```xml
    <repositories>
        <repository>
            <id>hamster-public</id>
            <url>http://maven.hamster3.cn/maven-public</url>
        </repository>
        <repository>
            <id>hamster-repo</id>
            <url>http://maven.hamster3.cn/hamster-repo</url>
        </repository>
    </repositories>
```

I don't like this, so I try to add a `hamster-repo` to proxied
```
proxied [
  http://localhost:80/hamster-repo
  https://hub.spigotmc.org/nexus/content/repositories/snapshots
]
```

However, I have received (a lot of) the following error reports. At the same time, `hamster-repo` artifact does appear in Maven public, but the download speed of dependency is dozens of times slower than before!

```
18:01:24.135 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.127 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.127 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.124 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.124 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.124 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.123 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.123 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
18:01:24.120 ERROR | Proxied repository http://localhost:80/hamster-repo is unavailable due to: Read timed out
```

Obviously, this is not feasible, so I come here for help. Can the author update the relevant code to add this function?

**Comment by MiniDay** [comment]
my English level is not very good, please forgive me if I offend you

**Comment by dzikoysk** [comment]
Hi there, don't worry, your English is pretty good! :) 

Well, I see 2 issues here. At this moment there is no possibility to proxy other repository to the specific one (see: #448). It's something what we want to add in 3.0, but it will take some time. 
The next issue is related to hardcoded timeouts and I think we can move it to the configuration to let people decide how long Reposilite can awaits for incoming data. I think I may add this today, in like about 3 hours.

**Comment by dzikoysk** [comment]
Here is the nightly version with timeouts in configuration 

* https://repo.panda-lang.org/snapshots/org/panda-lang/reposilite/2.9.22-SNAPSHOT/reposilite-2.9.22-20210503.121732-5.jar

Per repository proxying will be available in Reposilite 3.x

**Comment by MiniDay** [comment]
> Here is the nightly version with timeouts in configuration
> 
> * https://repo.panda-lang.org/snapshots/org/panda-lang/reposilite/2.9.22-SNAPSHOT/reposilite-2.9.22-20210503.121732-5.jar
> 
> Per repository proxying will be available in Reposilite 3.x

okay, thank you! I'm looking forward to the 3.0 release

**Comment by dzikoysk** [comment]
Addressed in #544
